pkg/proc/bininfo.go

```diff
@@ -224,6 +287,97 @@ func (bi *BinaryInfo) LoadError() error {
 	return bi.loadErr
 }
 
+type nilCloser struct{}
+
+func (c *nilCloser) Close() error { return nil }
+
+// New creates a new BinaryInfo object using the specified data. Use LoadBinary instead.
+func (bi *BinaryInfo) LoadFromData(dwdata *dwarf.Data, debugFrameBytes, debugLineBytes, debugLocBytes []byte) {
+	bi.closer = (*nilCloser)(nil)
+	bi.dwarf = dwdata
+
+	if debugFrameBytes != nil {
+		bi.frameEntries = frame.Parse(debugFrameBytes, frame.DwarfEndian(debugFrameBytes))
+	}
+
+	bi.loclistInit(debugLocBytes)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go bi.loadDebugInfoMaps(debugLineBytes, &wg)
```

Comment:
Maybe we modify this method to check if `wait group != nil` before calling done. That way here we don't have to go through the trouble of creating a wait group and spawning a goroutine.